src/main.cc

```diff
@@ -86,6 +87,28 @@ class Spellchecker : public Nan::ObjectWrap {
     }
   }
 
+  static NAN_METHOD(CheckSpellingCallback) {
+    Nan::HandleScope scope;
+    if (info.Length() < 2) {
+      return Nan::ThrowError("Bad argument");
+    }
+
+    Handle<String> string = Handle<String>::Cast(info[0]);
+    if (!string->IsString()) {
+      return Nan::ThrowError("Bad argument");
+    }
+
+    Nan::Callback *callback = new Nan::Callback(info[1].As<v8::Function>());
+
+    std::vector<uint16_t> *corpus = new std::vector<uint16_t>(string->Length() + 1);
+    string->Write(reinterpret_cast<uint16_t *>(corpus->data()));
```

Comment:
v8 Strings are immutable so I was assuming that it's safe to pass this std::vector into the worker. I was hoping to avoid copying the corpus data if at all possible so we can handle huge files without blowing out more RAM than we have to.